### PR TITLE
relations: added subject relation in awards & 📦 release: v15.0.0

### PR DESCRIPTION
**Pull request by 0einstein0**
:heart: Thank you for your contribution!

### Description

Closes Issue https://github.com/inveniosoftware/invenio-vocabularies/issues/403

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_rdm_records/records/dumpers/subject_hierarchy.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+            hierarchy.append(
+                f"{current_hierarchy}{self._splitchar}{current_subject_id}"
+            )
+            return hierarchy
+
+        for award in awards:
+            subjects = award.get("award", {}).get("subjects", [])
+            for subject in subjects:
+                parents = subject.get("props", {}).get("parents", "")
```
Comment (anikachurilova) [inline invenio_rdm_records/records/dumpers/subject_hierarchy.py:54]:
```suggestion
                parents = subject.get("props", {}).get("parents", [])
```

File: invenio_rdm_records/records/dumpers/subject_hierarchy.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                current_subject_id = subject.get("id", "")
+                if parents and current_subject_id:
+                    subject_hierarchy = build_hierarchy(parents, current_subject_id)
+                    subject.setdefault("props", {})["hierarchy"] = subject_hierarchy
+
+        data["metadata"] = data.get("metadata", {})
```
Comment (anikachurilova) [inline invenio_rdm_records/records/dumpers/subject_hierarchy.py:60]: Why is this needed?

File: invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json [hunk 3]
```diff
@@ -381,12 +348,55 @@
...
         "type": "string"
+      },
+      "scheme": {
+        "description": "Identifier of the subject scheme.",
+        "$ref": "local://definitions-v1.0.0.json#/identifier"
+      },
+      "props": {
+        "type": "object",
+        "patternProperties": {
+          "^.*$": {
```
Comment (anikachurilova) [inline invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json:359]: Since this is allowing any property, why is this needed?

File: invenio_rdm_records/records/dumpers/subject_hierarchy.py (hunk 2 above)
Comment (0einstein0) [reply invenio_rdm_records/records/dumpers/subject_hierarchy.py:60]: leftover, will remove and update

File: invenio_rdm_records/records/dumpers/subject_hierarchy.py (hunk 1 above)
Comment (anikachurilova) [reply invenio_rdm_records/records/dumpers/subject_hierarchy.py:54]: You are right, thanks for the explanation!

File: invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json (hunk 3 above)
Comment (0einstein0) [reply invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json:359]: Just for defining the constraint that the values for those keys must be strings, as we are storing parents as a string.
the subject hierarchy is a list but it is not stored anywhere just dumped when indexing.

File: invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json [hunk 4]
```diff
@@ -381,12 +348,55 @@
...
+      },
+      "organization": {
+        "description": "Human readable label.",
+        "type": "string"
+      }
+    }
+  },
+  "organizations": {
+    "type": "array",
+    "items": { "$ref": "#/organization" }
```
Comment (alejandromumo) [inline invenio_rdm_records/records/jsonschemas/records/definitions-v2.0.0.json:399]: Discussed IRL, these are likely not needed. 

DB-wise, subjects are linked in drafts / records either by their `id` (controlled vocabulary) or `subject` (custom subjects)

**Review by ptamarit — APPROVED** [review]
